### PR TITLE
make the docs of arithmetics better

### PR DESCRIPTION
**Pull request by ringabout**
Ref https://github.com/timotheecour/Nim/issues/415

code-blocks => `runnableExamples`
double backticks => single backticks

### CODE REVIEW

**Comment by ringabout** [comment]
`proc + noSideEffect` => func makes {.borrow.} wrong

File: lib/system/arithmetics.nim [hunk 1]
```diff
@@ -167,20 +167,19 @@ proc `-`*(x: int16): int16 {.magic: "UnaryMinusI", noSideEffect.}
...
+    var
+      a = 0'u8
+      b = 0'i8
+      c = 1000'u16
+      d = 1000'i16
+
+    assert not a == 255
+    assert not b == -1
+    assert not c == 64535
+    assert not d == -1001
```
Comment (timotheecour) [inline lib/system/arithmetics.nim]: inline variables used once; makes the test more readable
```suggestion
    assert not 0'u8 == 255
    assert not 0'i8 == -1
    assert not 1000'u16 == 64535
    assert not 1000'i16 == -1001
```

File: lib/system/arithmetics.nim [hunk 2]
```diff
@@ -1,48 +1,48 @@
...
+  ## If such a value does not exist, `OverflowDefect` is raised
   ## or a compile time error occurs.
+  runnableExamples:
+    let x = 5
+    assert succ(5) == 6
+    assert succ(5, 3) == 8
+
+proc pred*[T: Ordinal](x: T, y = 1): T {.magic: "Pred", noSideEffect.} =
+  ## Returns the `y`-th predecessor (default: 1) of the value `x`.
+  ## `T` has to be an `ordinal type <#Ordinal>`_.
```
Comment (timotheecour) [inline lib/system/arithmetics.nim]:
```suggestion
```

redundant with type declaration (and other procs don't do this)

File: lib/system/arithmetics.nim [hunk 3]
```diff
@@ -1,48 +1,48 @@
-proc succ*[T: Ordinal](x: T, y = 1): T {.magic: "Succ", noSideEffect.}
-  ## Returns the ``y``-th successor (default: 1) of the value ``x``.
-  ## ``T`` has to be an `ordinal type <#Ordinal>`_.
+proc succ*[T: Ordinal](x: T, y = 1): T {.magic: "Succ", noSideEffect.} =
+  ## Returns the `y`-th successor (default: 1) of the value `x`.
+  ## `T` has to be an `ordinal type <#Ordinal>`_.
```
Comment (timotheecour) [inline lib/system/arithmetics.nim]:
```suggestion
```

File: lib/system/arithmetics.nim [hunk 4]
```diff
@@ -1,48 +1,48 @@
...
   ##
-  ## If such a value does not exist, ``OverflowDefect`` is raised
+  ## If such a value does not exist, `OverflowDefect` is raised
   ## or a compile time error occurs.
+  runnableExamples:
+    let x = 5
```
Comment (timotheecour) [inline lib/system/arithmetics.nim]:
```suggestion
```

File: lib/system/arithmetics.nim [hunk 5]
```diff
@@ -1,48 +1,48 @@
...
-  ##
-  ## If such a value does not exist, ``OverflowDefect`` is raised or a compile
-  ## time error occurs. This is a short notation for: ``x = succ(x, y)``.
-  ##
-  ## .. code-block:: Nim
-  ##  var i = 2
-  ##  inc(i)    # i <- 3
-  ##  inc(i, 3) # i <- 6
+  runnableExamples:
+    let x = 5
```
Comment (timotheecour) [inline lib/system/arithmetics.nim]:
```suggestion
```

File: lib/system/arithmetics.nim [hunk 6]
```diff
@@ -207,34 +206,32 @@ proc `*`*(x, y: int16): int16 {.magic: "MulI", noSideEffect.}
...
-  ##
-  ## .. code-block:: Nim
-  ##   ( 1 div  2) ==  0
-  ##   ( 2 div  2) ==  1
-  ##   ( 3 div  2) ==  1
-  ##   ( 7 div  3) ==  2
-  ##   (-7 div  3) == -2
-  ##   ( 7 div -3) == -2
-  ##   (-7 div -3) ==  2
+  ## This is roughly the same as `trunc(x/y)`.
```
Comment (timotheecour) [inline lib/system/arithmetics.nim]:
```suggestion
  ## This is roughly the same as `math.trunc(x/y).int`.
```

**Review by timotheecour — APPROVED** [review]

**Pull request merged**